app/app.go

```diff
@@ -855,3 +857,30 @@ func (app *App) Close() (err error) {
 	app.BlockValidator.StopWait()
 	return nil
 }
+
+func (app *App) CheckTx(req abci.RequestCheckTx) abci.ResponseCheckTx {
```

Comment:
I would combine those two, and reverse the condition in this "if"

Reply:
After reversing the "if" condition, I would have to return `app.BaseApp.CheckTx(req)` twice, so I won't change it.